File: week05/week05-5.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        int a[100000], N=0;
        while(head!=NULL){
            a[N++] = head -> val;
            head = head -> next;
        }
        for(int i=0;i<N/2; i++){
            if(a[i] !=a[N-1-i]) return false;
        }
        return true;
    }
};

File: week05/week05-5.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        a = []
        while head!=None:
            a.append(head.val)
            head = head.next
        N = len(a)
        for i in range(N):
            if a[i] != a[N-1-i]: return False
        return True